File: packages/typia/src/programmers/internal/json_schema_boolean.ts
import { OpenApi } from "@samchon/openapi";

import { MetadataAtomic } from "../../schemas/metadata/MetadataAtomic";

import { json_schema_plugin } from "./json_schema_plugin";

export const json_schema_boolean = (
  atomic: MetadataAtomic,
): OpenApi.IJsonSchema.IBoolean[] =>
  json_schema_plugin({
    schema: {
      type: "boolean",
    } satisfies OpenApi.IJsonSchema.IBoolean,
    tags: atomic.tags,
  });

File: packages/typia/native/core/programmers/iterate/json_schema_boolean.go
package iterate

import nativemetadata "github.com/samchon/typia/packages/typia/native/core/schemas/metadata"

func json_schema_boolean(atomic *nativemetadata.MetadataAtomic) []JsonSchema {
	return json_schema_plugin(struct {
		schema JsonSchema
		tags   [][]nativemetadata.IMetadataTypeTag
	}{
		schema: JsonSchema{"type": "boolean"},
		tags:   atomic.Tags,
	})
}

func Json_schema_boolean_export(atomic *nativemetadata.MetadataAtomic) []JsonSchema {
	return json_schema_boolean(atomic)
}
